### ISSUE TITLE
AbortError: The user aborted a request

### ISSUE THREAD

**Issue by martinGITHUBER** [Jisagi/Discord-guild-copy#80]
**Please describe the problem you are having in detail:**
Every time I try to restore, at some point in the process, the script exits with a non-zero exit status code.

**Please describe the problem in as much detail as possible. Please include any thrown error codes as well. **
At some point during restoration(Not at any specific point), there is an AbortError: The user aborted a request

**Additional information:**
After doing some research, I discovered that this is probably because there are so many text channels in my guild, that discord.js just aborts the request. My suggestion would be to maybe do all the text channel restoration in chunks, because some guilds have a **lot** of text channels

- Node.js Version: v12.18.3
- Operating system: Debian 10(Buster). Raspberry Pi OS (32-bit) Lite

**Comment by Jisagi** [comment]
Sadly, the library doesn't allow any bulk create action. The creation can indeed be a problem. I do have an idea, but cannot verify if this would help. I could implement some kind of (configurable) timeout after each (channel) creation to stretch them a bit over time and maybe avoid a ratelimit. Usually the script should show a ratelimit error, but it doesn't in your case, so I'm not sure if this would even help or not.
For now, can you tell me how many channel (roughly) does the guild have youa re trying to clone/restore, so I can try to replicate this behaviour?

**Comment by martinGITHUBER** [comment]
51 Text Channels and 14 Voice channels

**Comment by Jisagi** [comment]
Ok, I tried to replicate the issue, but sadly I couldn't. I created a new server with ~100 text channel and ~50 voice channel and was able to restore it without problem. Usually the script only ha a "problem" with ratelimits and any solution I'd implement would only help with that, but nothing else.

I do remember the error you posted though. I did have this exact error randomly on my private bots. It happens when the library doesn't get an answer from the api within I think ~15s. This is **not** a ratelimit and therefore sadly nothing I can fix! Might be an api timeout on discords side or something else we can't really do anything against. There is still an open issue on the d.js library github for this error with the issue number 3471, but it sadly doesn't solve your issue either.

Anyway, I will still implement a configurable timeout for the creation on restore & clone, so large(r) guilds don't encounter ratelimits (that fast) anymore. Things like emojis and especially guild settings like its icon are still heavily ratelimited and shouldn't be changed too often. All this will probably not help you specifically, but it's the only thing I can help with in this situation.
